test/unit/org/apache/cassandra/cql3/validation/entities/VirtualTableTest.java

```diff
@@ -88,17 +89,45 @@ public void apply(PartitionUpdate update)
         }
     }
 
+    private abstract static class TruncatableVirtualTable extends AbstractVirtualTable
+    {
+        private boolean isTruncated = false;
```

Comment:
Virtual table can be called by multiple threads and that the tread reading the VT might be different from the one performing the truncation.

Reply:
Your comment is a bit confusing because I'm not sure whether you are proposing to change something or not. I totally agree on the fact that multiple threads can initiate the truncation. However, I see no point for adding any kind of thread safety here because it is just a test implementation. In fact, in tests there are no occasions of concurrent access to this VT.

Reply:
Sorry for being confusing. You are right with the way the test is done today there is no concurrency race.
I have seen in the past some tests where we decided to switch from `execute` to `executeNet` to do some extra testing for authorization for example. In such case some code become mutltithreaded and people my miss the fact that the test was not intended to support concurrency initially. To prevent such a problem, I would have made the variable `volatile` to be on the safe side.